crates/model/src/order.rs

```diff
@@ -352,8 +352,14 @@ impl OrderData {
     }
 
     /// Checks if the order is a market order.
-    pub fn within_market(&self, quote_sell_amount: &U256, quote_buy_amount: &U256) -> bool {
-        self.sell_amount.full_mul(*quote_buy_amount) >= quote_sell_amount.full_mul(self.buy_amount)
+    pub fn within_market(
+        &self,
+        quote_sell_amount: &U256,
+        quote_buy_amount: &U256,
+        quote_fee_amount: &U256,
```

Comment:
nit: in the other place it paid off to use a struct to avoid accidentally mixing up the different arguments

Reply:
added.